sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryClientBuilder.java

```diff
@@ -0,0 +1,327 @@
+        // Azure.Core deprecated setting the applicationId in httpLogOptions, but we should still support it.
+        String applicationId = clientOptions == null
+            ? httpLogOptions.getApplicationId()
+            : clientOptions.getApplicationId();
+
+        policies.add(new UserAgentPolicy(applicationId, clientName, clientVersion, configuration));
+
+        // Adds a "x-ms-client-request-id" header to each request. This header is useful for tracing requests through Azure ecosystems
+        policies.add(new RequestIdPolicy());
+
+        // Only the RequestIdPolicy  and UserAgentPolicy will take effect prior to the retry policy since neither of those need
+        // to change in any way upon retry
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+        policies.add(retryPolicy);
+
+        // Adds a date header to each HTTP request for tracking purposes
+        policies.add(new AddDatePolicy());
+
+        policies.addAll(additionalPolicies);
+
+        // If client options has headers configured, add a policy for each
+        if (clientOptions != null) {
+            List<HttpHeader> httpHeaderList = new ArrayList<>();
+            clientOptions.getHeaders().forEach(header ->
+                httpHeaderList.add(new HttpHeader(header.getName(), header.getValue())));
+            policies.add(new AddHeadersPolicy(new HttpHeaders(httpHeaderList)));
+        }
+
+        // Custom policies, authentication policy, and add date policy all take place after the retry policy which means
+        // they will be applied once per http request, and once for every retried http request. For instance, the
+        // AddDatePolicy will add a date time header for each request that is sent, and if the http request fails
+        // and the retry policy dictates that the request should be retried, then the date time header policy will
+        // be applied again and the current date time will be put in the header instead of the date time from
+        // the first http request attempt.
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+        policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+        return new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+    }
+
+    /**
+     * Create a {@link ModelsRepotioryClient} based on the builder settings.
+     *
+     * @return the created synchronous ModelsRepotioryClient
+     */
+    public ModelsRepotioryClient buildClient() {
+        return new ModelsRepotioryClient(buildAsyncClient());
+    }
+
+    /**
+     * Create a {@link ModelsRepositoryAsyncClient} based on the builder settings.
+     *
+     * @return the created asynchronous ModelsRepositoryAsyncClient
+     */
+    public ModelsRepositoryAsyncClient buildAsyncClient() {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
```

Comment:
I'm curious why there isn't any sort of TokenCredential or AzureSasCredential policy here. Does this service not have any sort of authentication? I can't find it in the .NET repo either

Reply:
There is no authentication support as of today. The global repository is also public and no credentials.

Reply:
Interesting. Got it!